feat: add component preview to library authoring [FC-0062]

rpenido opened the pull request:
## Description
This PR adds an XBlock preview to the library authoring sidebar.
![image](https://github.com/user-attachments/assets/2eded04f-fd6c-4483-a77d-b7b3cbe20156)

## More information
Part of:
 - https://github.com/openedx/frontend-app-course-authoring/issues/1189
 
Depends on:
- https://github.com/openedx/edx-platform/pull/35473

## Testing Instructions
- Checkout this branch and https://github.com/openedx/edx-platform/pull/35473
- Open the library authoring home for a library with some components
- Click on the component and check the preview on the sidebar. You can't interact with the XBlock here.
- Clicking on expand should open a modal dialog with the component preview. In this preview, you can interact with the XBlock
___
Private ref: [FAL-3801](https://tasks.opencraft.com/browse/FAL-3801)

rpenido commented on src/library-authoring/LibraryBlock/wrap.js:
```diff
@@ -0,0 +1,392 @@
+/* eslint-disable no-param-reassign */
+/* istanbul ignore file */
```
I'm not sure how we can test this, as a good part of this code relies on the hosted `xblock-bootstrap.html`.

bradenmacdonald replied:
Yeah, we can't really test it without actually having the LMS running. So just leave it for now.

ChrisChV commented on src/library-authoring/LibraryBlock/LibraryBlock.tsx:
```diff
@@ -0,0 +1,94 @@
+    }
+
+    const { method, replyKey, ...args } = event.data;
+    // const frame = iframeRef.current.contentWindow;
+    // const sendReply = async (data) => {
+    //   frame?.postMessage({ ...data, replyKey }, '*');
+    // };
```
I think we can delete this commented code?

rpenido replied:
Fixed here: 0ed8e025efc74ea8ebf0ddd8f4e2916e34f60fdc

ChrisChV commented on src/library-authoring/data/apiHooks.ts:
```diff
@@ -53,11 +53,10 @@ export const libraryAuthoringQueryKeys = {
     'libraryBlockTypes',
   ],
   xblockFields: (contentLibraryId: string, usageKey: string) => [
-    ...libraryAuthoringQueryKeys.all,
```
@rpenido Why is this being deleted? If the query `libraryAuthoringQueryKeys.all` is invalidated, then `xblockFields` is not invalidated.

rpenido replied:
Sorry. Not intended! This was fixed after I merged the last changes.

ChrisChV reviewed (approved):
@rpenido Looks good. Great work! Some small nits :+1:

One question. The `Expand` button overlaps the text on some components. is that expected?

![image](https://github.com/user-attachments/assets/e45eed3b-84ec-4317-9dae-b4233963c805)


- [X] I tested this: I followed the testing instructions and I tested with multiple component types.
- [X] I read through the code and considered the security, stability and performance implications of the changes.
- [X] I tested that the UI can be used with a keyboard only (tab order, keyboard controls).
- [X] Includes tests for bugfixes and/or features added.
- [X] Includes documentation
- [X] Includes fixtures that create objects required for manual testing.

rpenido commented:
Thanks for your review @ChrisChV!
I addressed the issues you found.

> One question. The `Expand` button overlaps the text on some components. is that expected?

Yes! I think this was the intention of the design.
We can't change it if it is an issue in the UI/UX review.

bradenmacdonald commented:
@rpenido At a very specific browser window height, there is some weird issue at the bottom of the modal:

![Screenshot](https://github.com/user-attachments/assets/f7e90388-df26-4072-ba5a-3b62c74a84f3)

Most other heights it appears fine but if you resize it, you'll see this issue. Also it seems weird that there are _two_ close buttons on the modal. Can we just remove the "bottom bar" of the modal (get rid of "Close", but keep the "X"), or was that specified by the UX team?

bradenmacdonald commented on src/library-authoring/LibraryBlock/LibraryBlock.tsx:
```diff
@@ -0,0 +1,94 @@
+  };
+
+  /**
+   * Load the XBlock data from the LMS and then inject it into our IFrame.
```
```suggestion
   * Prepare to receive messages from the IFrame.
```
We're not "injecting" the XBlock HTML/data anymore - the LMS is now rendering it.

bradenmacdonald commented on src/library-authoring/LibraryBlock/LibraryBlock.tsx:
```diff
@@ -0,0 +1,94 @@
+   * Load the XBlock data from the LMS and then inject it into our IFrame.
+   */
+  useEffect(() => {
+    // Prepare to receive messages from the IFrame.
```
```suggestion
```

rpenido commented:
> weird issue at the bottom of the modal

I think this is an issue with the Paragon modal introduced after changing the overflow behavior.
Tomorrow, I'll check other Paragon components to remove the Close button and see if it reduces the chance of this happening again.

bradenmacdonald commented on src/library-authoring/component-info/ComponentPreview.tsx:
```diff
@@ -0,0 +1,69 @@
+        >
+          {intl.formatMessage(messages.previewExpandButtonTitle)}
+        </Button>
+        <LibraryBlock usageKey={usageKey} />
```
In the future, it would be awesome to use something like [react-reverse-portal](https://github.com/httptoolkit/react-reverse-portal) to "re-parent" the `<LibraryBlock>` so that it _moves_ from the sidebar to the modal instead of being _copied_ into the modal. Then for example, you could start playing a video in the sidebar and "Expand" it and it would still be playing in the modal. Please mention that to the UX team as a possible future improvement after this merges (and post MVP).

rpenido replied:
Sure! I will mention that!

bradenmacdonald reviewed (approved):
Looks good! Just some minor comments. Ping me on Mattermost if/when you want me to hit merge.

rpenido commented:
Thank you for your review @bradenmacdonald.
I removed the footer close button.
I tried other modal components, but the issue in Paragon affects all of them. :disappointed: 

I think this is ready to merge!
